### DISCUSSION TITLE
Can I use include syntax in Jinja2 template?

### DISCUSSION THREAD

**Discussion by joechiu-uipath** [microsoft/promptflow#1566] [General]
I have 2 prompts that have some overlapping text and instead of duplicating them, I would like to use jinja's include syntax.
But it seems to fail with the error message below, and online search seems to point to the Jinja2 template engine needs to be initialize with some loader parameter to be able to support include.

Any suggestion on how to manage multiple prompts sharing block of verbatim? Can promptflow's jinja2 engine support include?

<code>
Flow test failed with UserErrorException: Exception: Failed to render jinja template: TypeError: no loader for this environment specified. Please modify your prompt to fix the issue. 
</code>

**Comment by brynn-code** [comment]
I'm afraid promptflow not support that for now, as you mentioned that jinja2 template needs to be initialized with some specific parameters. For your case, is it possible to extract the common part to a prompt node? the prompt node has content like:
```
..common parts...
{{common_params}}
```
And this prompt node links to llm node. the llm node prompt looks like:
```
{{common_prompt}}
{{llm1_param}}
```
Then the flow may shows like:
![image](https://github.com/microsoft/promptflow/assets/24237253/23d535d2-8bd5-459a-a35b-f74d9d4a2a80)
